Zend/zend_compile.c

```diff
@@ -2974,7 +2988,234 @@ void zend_compile_static_call(znode *result, zend_ast *ast, uint32_t type) /* {{
 }
 /* }}} */
 
-void zend_compile_new(znode *result, zend_ast *ast) /* {{{ */
+zend_string* zend_name_anon_class(zend_ast *parent TSRMLS_DC) {
+    size_t len;
+    char   *val;
+    zend_string *anon;
+    uint32_t next = get_next_op_number(CG(active_op_array));
+    
+    if (parent) {
```

Comment:
Can this conditional be true? It is called with `decl->name` of the anon class, which is presumably always NULL.